images/rook/Dockerfile

```diff
@@ -16,5 +16,16 @@ FROM BASEIMAGE
 
 ADD rook rookflex /usr/local/bin/
 
+# Setup user for build execution and application runtime
+ENV ROOK_ROOT=/
+ENV PATH=${ROOK_ROOT}/bin:${PATH} HOME=${ROOK_ROOT}
+RUN chmod -R u+x /usr/local/bin/rook /usr/local/bin/rookflex && \
+    chgrp -R 0 ${ROOK_ROOT} && \
+    chmod -R g=u ${ROOK_ROOT} /etc/passwd
```

Comment:
`dataDirHostPath` is insignificant since its the path on the host. The more relevant path for us is [this](https://github.com/rook/rook/blob/master/cmd/rook/mon.go#L48) one which is inside the container. If this was fixed to `/var/lib/rook` then the only perm change we need it `chmod -R g=u /var/lib` or `chmod -R g+w /var/lib` which effectively gives write perms on `/var/lib` so that rook can `mkdir -p /var/lib/rook`.

However since the dataDir can change using the `config-dir` option we probably need a better fix. @travisn can we remove the `config-dir` option? It seems like this was more useful when we support `rook` running outside of a container.

I'm concerned about `chmod -R g=u /`.

Reply:
Perhaps we can create `/var/lib/ceph` during the build sequence (hardcoded) and then only do a `chmod -R g=u /var/lib/ceph` on runtime. `config-dir` can go away if we stick with ceph's default `/var/lib/ceph`.
Also `ROOK_ROOT` could be `/var/lib/ceph`?

Reply:
I think we need to leave room for other backends and rook specific files so `/var/lib/rook` make more sense.

Ceph could go in in `/var/lib/rook/ceph` which would keep everything in  one directory and `dataDirHostPath` continues to work as is. `HOME` could also be `/var/lib/rook`

Reply:
Why don't we have `/etc/rook` for Rook's config files, `/var/lib/ceph` for Ceph data and `/etc/ceph` for Ceph's config files?
With that, we can remove `config-dir` entirely.

Reply:
I'm also considering the case of having other backends besides ceph, for example, /var/lib/rook/ceph and /var/lib/rook/gluster. would be nice to have a single root directory for both such that we won't need multiple `dataDirHostPath`.

Reply:
I'm thinking of something like: https://github.com/rook/rook/pull/1362/commits/64e87ed5a6aa1d3b8723430ba4cc9bdfd48cc81b

Reply:
Somewhat OT: will this path become moot/obsolete anyway once we switch to using the kube managed local dirs?  (i'm forgetting what they're called).

Reply:
I don't mind having data and conf in `/var/lib/rook/ceph`, it's just really confusing for Ceph users.

@liewegas dunno

Reply:
I think we would still need a local path even when using K8S local storage. I agree that it would be nice to find a way to make /var/lib/ceph work as well. what if we create a symlink from `/var/lib/ceph` to `/var/lib/rook/ceph`?

Reply:
@bassam for `dataDirHostPath` yes I think having `/var/lib/rook/ceph/{lib,etc}` with their respective paths to `/var/lib/ceph` and `/etc/ceph` is good.

In the end, we can have something like `/var/lib/rook/$BACKEND/`.
Those links will be generated in the image depending on the chosen backend, so, for now, we will stick with Ceph only but leave room for other backends.

Are we on the same page?

Reply:
Is each cluster actually in $basedir/$namespace/?  So it would actually be /var/lib/rook/ceph/$namespace/{osd,mds,mon}/...?

It's already a bit different with e.g. ceph-helm (/var/lib/ceph/$namespace/{mon,osd,...}/, but at least it's findable.  That would be my preference.  Any reason it can't be backend-specific, so that  eventually gluster or whatever would also land there instead of living under a rook/ dir somewhere?  Rook feels like it should be owning the orchestration/operation but having it "own" the data directory too is odd.

Reply:
@leseb actually /var/lib/rook/ceph/$namespace/{lib,etc}, right?

Reply:
@liewegas is namespace necessary? A given container/pod will only be part of a single namespace?

> Any reason it can't be backend-specific, so that eventually gluster or whatever would also land there instead of living under a rook/ dir somewhere? Rook feels like it should be owning the orchestration/operation but having it "own" the data directory too is odd.

the only reason I see is having a single root dir to bind mount to the host via `dataDirHostPath` vs. multiple dirs that need to be mounted.

Reply:
Oh, I thought we were talking about the *host* path here (dataDir*Host*Path)?

Inside the container, it definitely needs to be /etc/ceph and /var/lib/ceph.  Bonus points if it's not a symlink, but that's mostly paranoia--I won't want to worry about tools or package pre/postinst scripts being confused.  If it's only those two dirs that get bind mounted (instead of the one) that doesn't seem any more complex than also having to go add symlinks. Would the symlinks also cause a problem since those directories would already exist inside the container image?  (*Can* you bind mount a symlink on top of a directory?)

Reply:
opened #1368 to capture this discussion.